23-04-2015/look-say/look-say-ankur.txt

```diff
@@ -0,0 +1,14 @@
+class LookSay
+  def self.next(num,upto=1)
+    arr_num = []
+    upto.times do |i|
+      arr_num[i] = (i == 0) ? lookandsay(num.to_s) : lookandsay(arr_num[i-1])
+    end
+    arr_num.map!(&:to_i)
+    arr_num.size == 1 ? arr_num.first : arr_num
+  end
+
+  def self.lookandsay(str)
+    str.gsub(/(.)\1*/) {$&.length.to_s + $1}
```

Comment:
Suppose str = '1121'

gsub(/(.)\1*/)

If want to repeat group(1) another time. use \1 and () represents group. So it repeats 1 until it gets 2.

Then it passes this block to {$&.length.to_s + $1}

So in the `& ` it gets this block and it counts the length of it. $1 is a global variable which can be used in later code. Here it stores the number last computed and it appends to the count. So we have **21** so we have 2 as the count of the previous number and 1 as the previous number. And by the substitution the group with the calculated value.